### PR TITLE
deal with fallthrough instances 

### PR DESCRIPTION
**Pull request by oberon2007**
- the occurrence in [timesig.cpp] is apparently unintentional as already discussed with @Jojo-Schmitz .
- the one in [pitchspellin.cpp] I think should break aswell, possibly also in [preferences].
- commented the other occurrences, which will suppress warnings by gcc7 and should also work for other newer compilers

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
I believe the fallthoughs in pitchspelling and prefences are intentional.

File: libmscore/keysig.cpp [hunk 1]
```diff
@@ -203,19 +203,31 @@ void KeySig::layout()
...
             case -5: addLayout(SymId::accidentalFlat, xo + 4.0 * fspread, lines[11]);
+                     // fall through
             case -4: addLayout(SymId::accidentalFlat, xo + 3.0 * fspread, lines[10]);
+                     // fall through
             case -3: addLayout(SymId::accidentalFlat, xo + 2.0 * fspread, lines[9]);
+                     // fall through
             case -2: addLayout(SymId::accidentalFlat, xo + 1.0 * fspread, lines[8]);
+                     // fall through
             case -1: addLayout(SymId::accidentalFlat, xo,                 lines[7]);
             case 0:
```
Comment (Jojo-Schmitz) [inline libmscore/keysig.cpp]: here's another fallthru, not marked as such?

Comment (oberon2007) [inline libmscore/keysig.cpp]: It's one of those not bothering gcc for some reason - will mark it anyway. There was another one somewhere else - I'll look for it again.

File: libmscore/pitchspelling.cpp [hunk 2]
```diff
@@ -725,9 +725,11 @@ void spell(std::vector<Event>& notes, int key)
                         case 3:
                               k = end - start - 3;
                               notes[end-3].setTpc(tab[(notes[end-3].dataA() % 12) * 2 + ((opt & (1<<k)) >> k)]);
+                              break;
```
Comment (lasconic) [inline libmscore/pitchspelling.cpp]: I think the fallback is intended here. @wschweer could you confirm?

**Comment by lasconic** [comment]
I cherrypicked all commits but the one for pitch spelling.

**Comment by Jojo-Schmitz** [comment]
Sure about the changes in preferences.cpp?

**Comment by lasconic** [comment]
There is no change in preferences.cpp :)

**Comment by Jojo-Schmitz** [comment]
Ah, forget it there is does seem correct to fall through.

**Pull request closed without merging**